StarRocks/starrocks
# [BugFix] check if helper in local before bdb setup

## What type of PR is this：

## Which issues of this PR fixes ：

Fixes #7498 

## Problem Summary(Required) ：

How to reproduce this problem:
1. Deploy a cluster with 1 master + 2 followers.
2. Restart the current master, wait until the whole cluster is stable. Let's say the new master is node A.
3. Add a new node X to the cluster, but start without `--helper` argument. Node X will restart successfully and consider itself as a master of its own cluster.
4. Restart the node X with `--helper` argument. 

We expect this time node X will be down for failure to join the cluster, but it turns out that node A is down. Which means that add a bad follower to a cluster could leading to master down.

The main reason for this bug is a bad design in the BDBJE library. When a node X joins an existing cluster, there is no mechanism to check if the cluster information in the metadata of node X matches the old ones such as node A.

Node X will vote for itself as the new master right after the connection is built, yet only when the id of node X is identical to node A will an exception throws at the side of node X. In other cases, node A will crash because of the conflict of ids when receives this proposal. Since node X is the first instance of its cluster, the id of node X is 1. But node A is not the first instance of its cluster, as a result node X successfully join the cluster and node A die of conflict of ids when handling the proposal from node X.

To solve this problem, we have to add a mechanism to check if local cluster matches the helper node. I add an extra step to init a observer environment so that we can read local nodes and check if they contain the helper node.
